src/chains/ethereum/ethereum/src/forking/handlers/base-handler.ts

```diff
@@ -135,41 +141,96 @@ export class BaseHandler {
     }
   }
 
-  getFromCache<T>(key: string) {
+  getFromMemCache<T>(key: string) {
     const cachedRequest = this.requestCache.get(key);
     if (cachedRequest !== undefined) return cachedRequest as Promise<T>;
 
     const cachedValue = this.valueCache.get(key);
     if (cachedValue !== undefined) return JSON.parse(cachedValue).result as T;
   }
 
+  async getFromSlowCache<T>(method: string, params: any[], key: string) {
+    if (!this.persistentCache) return;
+    const raw = await this.persistentCache.get(method, params, key).catch(e => {
+      if (e.notFound) return null;
+      // I/O or other error, throw as things are getting weird and the cache may
+      // have lost integrity
+      throw e;
+    });
+    if (raw !== undefined) return { result: JSON.parse(raw).result as T, raw };
+  }
+
   async queueRequest<T>(
+    method: string,
+    params: any[],
     key: string,
     send: (
       ...args: unknown[]
     ) => Promise<{
       response: { result: any } | { error: { message: string; code: number } };
       raw: string | Buffer;
-    }>
+    }>,
+    options = { noCache: false }
   ): Promise<T> {
-    const cached = this.getFromCache<T>(key);
-    if (cached !== undefined) return cached;
-
-    const promise = this.limiter.handle(send).then(({ response, raw }) => {
-      if (this.abortSignal.aborted) return Promise.reject(new AbortError());
+    if (!options.noCache) {
+      const memCached = this.getFromMemCache<T>(key);
+      if (memCached !== undefined) return memCached;
+
+      const diskCached = await this.getFromSlowCache<T>(method, params, key);
+      if (diskCached !== undefined) {
+        this.valueCache.set(key, Buffer.from(diskCached.raw));
```

Comment:
I think this line has a chance of throwing. `this.getFromSlowCache` can return `null` in certain cases. So I think line 180 should be `if(diskCached != undefined) {`

Reply:
Hm. I can't find where `getFromSlowCache` can return `null`